code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -347,8 +352,59 @@
 	modes = list(MODE_CREW)
 	var/target = null //for targeting in processing
 	var/target_set = FALSE //have we set a target at any point?
+	///Var to track the linked detective gun
+	var/linked_gun_UID
+
+/obj/item/pinpointer/crew/Initialize(mapload)
+	. = ..()
+	var/obj/item/gun/energy/detective/gun_to_link = locate(/obj/item/gun/energy/detective) in get_turf(src)
+	if(gun_to_link)
+		link_gun(gun_to_link.UID())
+
+/obj/item/pinpointer/crew/Destroy()
+	if(linked_gun_UID)
+		UnregisterSignal(locateUID(linked_gun_UID), COMSIG_DETGUN_TRACKING)
+	return ..()
+
+/obj/item/pinpointer/crew/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I, /obj/item/gun/energy/detective))
+		link_gun(I.UID())
+
+/obj/item/pinpointer/crew/proc/link_gun(gun_UID)
+	var/obj/item/gun/energy/detective/D = locateUID(gun_UID)
+	if(D.linked_pinpointer_UID || linked_gun_UID)
+		visible_message("<span class='notice'>The pinpointer pings to indicate either it or the gun is already linked.</span>", "<span class='notice'>You hear a pinpointer pinging.</span>")
+		return
+	D.link_pinpointer(UID())
+	linked_gun_UID = gun_UID
+	RegisterSignal(locateUID(linked_gun_UID), COMSIG_DETGUN_TRACKING, .proc/start_tracking, TRUE)
+
+/obj/item/pinpointer/crew/proc/start_tracking()
+	if(!linked_gun_UID)
+		return
+	var/obj/item/gun/energy/detective/D = locateUID(linked_gun_UID)
+	var/target_UID = D.tracking_target_UID
+	target = locateUID(target_UID)
+	target_set = TRUE
+	mode = MODE_DET
+	visible_message("<span class='notice'>The pinpointer flickers as it begins tracking a target relayed from a detective's revolver.</span>", "<span class='notice'>You hear a pinpointer flickering.</span>")
+	addtimer(CALLBACK(src, .proc/stop_tracking), 1 MINUTES, TIMER_UNIQUE)
+	START_PROCESSING(SSfastprocess, src)
```

Comment:
if you do continue to use the signal, procs called by a signal need `SIGNAL_HANDLER` at the top of them.